packages/ui-tailwind/src/utils/color.ts

```diff
@@ -0,0 +1,117 @@
+import { theme } from '../theme/theme';
+
+export type Colors = (typeof theme)['color'];
+
+/**
+ * This types gets a two-level-nested object of colors `e.g. { primary: { light: '#fff' } }`
+ * and transforms it to the union type of object keys with dot notation like `primary.light`.
+ */
+type MapColorsToDotNotation<Obj extends Record<string, Record<string, string>>> = {
+  [K1 in keyof Obj]: `${K1 & string}.${keyof Obj[K1] & string}`;
+}[keyof Obj];
+
+/**
+ * A union type of all theme colors. It is supposed to simplify passing color props to components
+ * without letting users provide generic classes.
+ */
+export type ThemeColor = MapColorsToDotNotation<Colors>;
+
+/**
+ * Takes a color string in the format of `primary.light` and returns the color value from the theme.
+ */
+export const getThemeColor = (color: ThemeColor): string => {
+  const colors = theme.color;
+
+  try {
+    const [group, shade] = color.split('.') as ['primary', keyof Colors['primary']];
+    return colors[group][shade];
+  } catch (_) {
+    throw new Error(`Color "${color}" not found in the registered Tailwind theme`);
+  }
+};
+
+/** This mapper class is needed to help Tailwind statically analyze the classes that could
+ * be produced from the `getThemeColorClass` function. */
+export const COLOR_CLASS_MAP: Record<ThemeColor, [string, string]> = {
+  'neutral.main': ['text-neutral-main', 'bg-neutral-main'],
+  'neutral.light': ['text-neutral-light', 'bg-neutral-light'],
```

Comment:
this color mapping is needed for Tailwind's compiler. It needs to see static classes, dynamic ones fail to be parsed